controllers/database_reconciler_helpers.go

```diff
@@ -70,15 +70,19 @@ func (r *DatabaseReconciler) addFinalizer(ctx context.Context, req ctrl.Request,
 func (r *DatabaseReconciler) handleDelete(ctx context.Context, database *ndbv1alpha1.Database, ndbClient *ndb_client.NDBClient) (ctrl.Result, error) {
 	log := ctrllog.FromContext(ctx)
 	log.Info("Database CR is being deleted.")
+	r.recorder.Event(database, "Normal", "DatabaseCRDeletionStarted", "Database CR is being deleted.")
 	if controllerutil.ContainsFinalizer(database, common.FINALIZER_DATABASE_INSTANCE) {
 		// Check if the database instance id (database.Status.Id) is present in the status
 		// If present, then make a deprovisionDatabase API call to NDB
 		// else proceed with removing finalizer as database instance provisioning wasn't successful earlier.
 		if database.Status.Id != "" {
 			log.Info("Deprovisioning database instance from NDB.")
+			r.recorder.Event(database, "Normal", "DatabaseDeprovisioningStarted", "Deprovisioning database instance from NDB.")
 			_, err := ndb_api.DeprovisionDatabase(ctx, ndbClient, database.Status.Id, *ndb_api.GenerateDeprovisionDatabaseRequest())
 			if err != nil {
-				log.Error(err, "Deprovisioning database instance request failed.")
+				errStatement := "Deprovisioning database instance request failed."
+				log.Error(err, errStatement)
+				r.recorder.Event(database, "Warning", "DatabaseDeprovisioningError", errStatement)
```

Comment:
Shouldn't it be error instead?

Reply:
events in k8s only have a 'Warning' type @manavrajvanshi. there is no 'Error' type.